### PR TITLE
Added VerifySsl option

### LINKED ISSUE #133: Allow to ignore SSL certificate validation
add an option for skipping SSL validation

`curl_easy_setopt(curl, CURLOPT_SSL_VERIFYPEER, 0L);`

### PR DESCRIPTION
**Pull request by hdak1945git**
This branch implements #95 (and possibly fixes #133 and #137)

---

The VerifySsl class is simply a wrapper around one boolean value; actionable code is located in `session.cpp`.

Removed all but one instance of INSECURE_CURL; which is located in: `test/proxy_tests.cpp`

### CODE REVIEW

File: cpr/session.cpp [hunk 1]
```diff
@@ -286,6 +283,14 @@ void Session::Impl::SetLowSpeed(const LowSpeed& low_speed) {
     }
 }
 
+void Session::Impl::SetVerifySsl(const VerifySsl& verify) {
+    auto curl = curl_->handle;
+    if (curl) {
+        curl_easy_setopt(curl, CURLOPT_SSL_VERIFYPEER, verify ? 1L : 0L);
+        curl_easy_setopt(curl, CURLOPT_SSL_VERIFYHOST, verify ? 2L : 0L);
```
Comment (whoshuu) [inline cpr/session.cpp]: Great work getting the defaults right. They're not very obvious unless you dig into the docs.

**Comment by hdak1945git** [comment]
@whoshuu Didn't notice your other commits until just recently, re-based and re-submitted. Hopefully all is set now. 

(Thanks for the note about `CURLOPT_SSL_VERIFYHOST` -- I was surprised when I read the documentation on that...)

File: cpr/ssl_options.cpp [hunk 2]
```diff
@@ -0,0 +1,12 @@
+#include "cpr/ssl_options.h"
+
+namespace cpr {
+
+VerifySsl::VerifySsl(bool verify)
+          : verify_{verify} {}
```
Comment (whoshuu) [inline cpr/ssl_options.cpp]: Can you put this on the above line? The format checker is [complaining](https://travis-ci.org/whoshuu/cpr/jobs/192078769#L7879-L7880). I'll write up instructions for how to do this manually at some point, and to also fail the build if the format checker doesn't pass.

Comment (hdak1945git) [inline cpr/ssl_options.cpp]: I ran clang-format over the offending files and resubmitted; I assume that's what you wanted.

**Comment by whoshuu** [comment]
Merged in https://github.com/whoshuu/cpr/commit/67d0b59e2eb3887afb4af3d34444f2ee32968c07. Thanks @hdak1945git!

**Pull request closed without merging**